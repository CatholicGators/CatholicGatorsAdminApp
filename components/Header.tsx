import React from 'react';

import { withStyles, createStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import AccountCircle from '@material-ui/icons/AccountCircle';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import Button from '@material-ui/core/Button';
import { User } from 'lib/withAuth';

const styles = createStyles({
    root: {
        flexGrow: 1,
    },
    grow: {
        flexGrow: 1,
    },
    menuButton: {
        marginLeft: -12,
        marginRight: 20,
    },
});

interface HeaderProps {
    user: User;
    classes: any;
}

class Header extends React.Component {
    state = {
        anchorEl: null,
    };

    constructor(public props: HeaderProps) {
        super(props);
    }

    handleMenu(event) {
        this.setState({ anchorEl: event.currentTarget });
    };

    handleClose() {
        this.setState({ anchorEl: null });
    };

    render() {
        const { user, classes } = this.props;
        const { anchorEl } = this.state;

        return (
            <div className={classes.root}>
                <AppBar position="static">
                    <Toolbar>
                        <IconButton className={classes.menuButton} color="inherit" aria-label="Menu">
                            <MenuIcon />
                        </IconButton>
                        <Typography variant="h6" color="inherit" className={classes.grow}>
                            Catholic Gators Admin
                        </Typography>
                        {user ? (
                            <div>
                                <IconButton
                                    aria-owns={anchorEl ? 'menu-appbar' : undefined}
                                    aria-haspopup="true"
                                    onClick={this.handleMenu.bind(this)}
                                    color="inherit"
                                >
                                    <AccountCircle />
                                </IconButton>
                                <Menu
                                    id="menu-appbar"
                                    anchorEl={anchorEl}
                                    anchorOrigin={{
                                        vertical: 'top',
                                        horizontal: 'right',
                                    }}
                                    transformOrigin={{
                                        vertical: 'top',
                                        horizontal: 'right',
                                    }}
                                    open={anchorEl}
                                    onClose={this.handleClose.bind(this)}
                                >
                                    <MenuItem onClick={this.handleClose.bind(this)}>Profile</MenuItem>
                                    <MenuItem onClick={this.handleClose.bind(this)}>My account</MenuItem>
                                </Menu>
                            </div>
                        ) : (
                            <Button color="inherit">Login</Button>
                        )}
                    </Toolbar>
                </AppBar>
            </div>
        );
    }
}

export default withStyles(styles)(Header);
