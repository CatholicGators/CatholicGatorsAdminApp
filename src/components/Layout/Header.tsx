import React from 'react';

import { withStyles, createStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Avatar from '@material-ui/core/Avatar';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import Button from '@material-ui/core/Button';

import withAuthentication from '../Auth/withAuthentication';

const styles = createStyles({
    root: {
        flexGrow: 1,
    },
    grow: {
        flexGrow: 1,
    },
    menuButton: {
        marginLeft: -12,
        marginRight: 20,
    },
});

class Header extends React.Component {
    state = {
        anchorEl: null,
    };

    constructor(public props: any) {
        super(props);
    }

    handleMenu(event) {
        this.setState({ anchorEl: event.currentTarget });
    }

    handleClose() {
        this.setState({ anchorEl: null });
    }

    handleLogin() {
        this.props.googleSignIn();
    }

    handleLogout() {
        this.props.signOut();
        this.handleClose();
    }

    render() {
        const { user, classes, children } = this.props;
        const { anchorEl } = this.state;

        return (
            <div className={classes.root}>
                <AppBar position="static">
                    <Toolbar>
                        <IconButton className={classes.menuButton} color="inherit" aria-label="Menu">
                            <MenuIcon />
                        </IconButton>
                        <Typography variant="h6" color="inherit" className={classes.grow}>
                            Catholic Gators Admin
                        </Typography>
                        {user ? (
                            <div>
                                <IconButton
                                    aria-owns={anchorEl ? 'menu-appbar' : undefined}
                                    aria-haspopup="true"
                                    onClick={this.handleMenu.bind(this)}
                                    color="inherit"
                                >
                                    <Avatar src={user.photoURL} />
                                </IconButton>
                                <Menu
                                    id="menu-appbar"
                                    anchorEl={anchorEl}
                                    anchorOrigin={{
                                        vertical: 'top',
                                        horizontal: 'right',
                                    }}
                                    transformOrigin={{
                                        vertical: 'top',
                                        horizontal: 'right',
                                    }}
                                    open={!!anchorEl}
                                    onClose={this.handleClose.bind(this)}
                                >
                                    <MenuItem onClick={this.handleLogout.bind(this)}>Logout</MenuItem>
                                </Menu>
                            </div>
                        ) : (
                            <Button color="inherit" onClick={this.handleLogin.bind(this)}>Login</Button>
                        )}
                    </Toolbar>
                </AppBar>
                {children}
            </div>
        );
    }
}

export default withAuthentication(withStyles(styles)(Header));