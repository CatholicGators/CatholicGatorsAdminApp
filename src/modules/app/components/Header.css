.root, .grow {
    flex-grow: 1;
}

.menuButton {
    margin-left: -12;
    margin-right: 20;
}