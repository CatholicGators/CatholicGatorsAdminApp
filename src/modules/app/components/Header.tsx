import React from 'react';
import { connect } from 'react-redux';
import { createStyles, withStyles } from '@material-ui/core';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Avatar from '@material-ui/core/Avatar';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import Button from '@material-ui/core/Button';
// import CircularProgress from '@material-ui/core/CircularProgress';

import { googleSignIn, signOut } from '../../../redux/actions/auth/authActions';

const styles = createStyles({
    root: {
        flexGrow: 1
    },
    grow: {
        flexGrow: 1
    },
    menuButton: {
        marginLeft: -12,
        marginRight: 20
    }
});

type Props = {
    user: any,
    signOut: () => any,
    googleSignIn: () => any
}

export class Header extends React.Component<Props> {
    state = {
        anchorEl: null,
    };

    constructor(public props) {
        super(props);
    }

    handleMenu(event) {
        this.setState({ anchorEl: event.currentTarget });
    }

    handleClose() {
        this.setState({ anchorEl: null });
    }

    handleLogin() {
        this.props.googleSignIn();
    }

    handleLogout() {
        this.props.signOut();
        this.handleClose();
    }

    profile(user) {
        const { anchorEl } = this.state;

        switch(user) {
            case undefined:
                return <div>Loading...</div>; // TODO: Replace this with CircularProgress

            case null:
                return <Button id="login-btn" color="inherit" onClick={this.handleLogin.bind(this)}>Login</Button>

            default:
                return <div>
                    <IconButton
                        id="avatar-btn"
                        aria-owns={anchorEl ? 'menu' : undefined}
                        aria-haspopup="true"
                        onClick={this.handleMenu.bind(this)}
                        color="inherit"
                    >
                        <Avatar src={user.photoURL} />
                    </IconButton>
                    <Menu
                        id="menu"
                        anchorEl={anchorEl}
                        anchorOrigin={{
                            vertical: 'top',
                            horizontal: 'right',
                        }}
                        transformOrigin={{
                            vertical: 'top',
                            horizontal: 'right',
                        }}
                        open={!!anchorEl}
                        onClose={this.handleClose.bind(this)}
                    >
                        <MenuItem id="logout" onClick={this.handleLogout.bind(this)}>Logout</MenuItem>
                    </Menu>
                </div>
        }
    }

    render() {
        const { user, classes } = this.props;
        // const { anchorEl } = this.state;

        return (
            <div className={classes ? classes.root : null}>
                <AppBar position="static">
                    <Toolbar>
                        <IconButton className={classes ? classes.menuButton : null} color="inherit" aria-label="Menu">
                            <MenuIcon />
                        </IconButton>
                        <Typography variant="h6" color="inherit" className={classes ? classes.grow : null}>
                            Catholic Gators Admin
                        </Typography>
                        {this.profile(user)}
                    </Toolbar>
                </AppBar>
            </div>
        );
    }
}

const mapStateToProps = state => ({
    user: state.auth.user
});

const mapDispatchToProps = dispatch => ({
    googleSignIn: () => dispatch(googleSignIn()),
    signOut: () => dispatch(signOut())
});

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(Header));
