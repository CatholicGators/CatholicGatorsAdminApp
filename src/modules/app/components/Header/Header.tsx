import React from 'react';
import { connect } from 'react-redux';
import { withRouter, NavLink } from "react-router-dom";

import { createStyles, withStyles, Theme } from '@material-ui/core';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import VpnKey from '@material-ui/icons/VpnKey';
import FormatAlignLeft from '@material-ui/icons/FormatAlignLeft';
import Inbox from '@material-ui/icons/Inbox';

import { googleSignIn, signOut } from '../../../../redux/actions/auth/authActions';
import MobileDrawer from '../MobileDrawer/MobileDrawer';
import ToolbarAvatar from '../ToolbarAvatar/ToolbarAvatar';

const styles = (theme: Theme) => createStyles({
    root: {
        flexGrow: 1
    },
    grow: {
        flexGrow: 1
    },
    menuButton: {
        marginLeft: -12,
        marginRight: 20,
        [theme.breakpoints.up('md')]: {
            display: 'none'
        }
    },
    toolbar: {
        minHeight: '64px'
    },
    desktopMenuItems: {
        display: 'flex',
        [theme.breakpoints.down('sm')]: {
            display: 'none'
        }
    },
    navLinkDesktop: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        paddingLeft: 20,
        paddingRight: 20,
        color: 'inherit',
        textDecoration: 'none',
        '&:visited': {
            color: 'inherit'
        },
        '&:hover': {
            color: theme.palette.secondary.main,
            transition: 'color',
            transitionDuration: '200ms',
        }
    },
    navLinkTextDesktop: {
        color: 'inherit'
    },
    desktopSelected: {
        color: theme.palette.secondary.main
    }
});

type Props = {
    user: any,
    signOut: () => any,
    googleSignIn: () => any
}

type State = {
    drawerOpen: boolean
}

export type MenuLink = {
    text: string,
    href: string,
    icon: any,
    needsAuthorization: boolean
}
export const menuLinks: Array<MenuLink> = [
    {
        text: 'Contact Form',
        href: '/',
        icon: FormatAlignLeft,
        needsAuthorization: false
    },
    {
        text: 'My Contacts',
        href: '/my-contacts',
        icon: Inbox,
        needsAuthorization: false
    },
    {
        text: 'Admin',
        href: '/admin',
        icon: VpnKey,
        needsAuthorization: true
    }
];

export class Header extends React.Component<Props, State> {
    state = {
        drawerOpen: false
    };

    constructor(public props) {
        super(props);
    }

    toggleDrawer(isOpen: boolean){
      this.setState({
        drawerOpen: isOpen,
      });
    };

    render() {
        const { classes, user } = this.props;

        return (
            <div className={classes ? classes.root : null}>
                <AppBar position="static">
                    <Toolbar className={classes ? classes.toolbar : null}>
                        <IconButton
                            id='menu-btn'
                            className={classes ? classes.menuButton : null}
                            color="inherit"
                            aria-label="Menu"
                            onClick={() => this.toggleDrawer(true)}
                        >
                            <MenuIcon/>
                        </IconButton>
                        <Typography variant="h6" color="inherit" className={classes ? classes.grow : null}>
                            Catholic Gators Admin
                        </Typography>
                        <div className={classes ? classes.desktopMenuItems : null}>
                            { user ?
                                menuLinks
                                    .filter(link => !link.needsAuthorization || (user.isAdmin && user.isApproved))
                                    .map(link => 
                                            <NavLink
                                                exact
                                                key={link.text}
                                                to={link.href}
                                                className={classes ? classes.navLinkDesktop : null}
                                                activeClassName={classes ? classes.desktopSelected : null}
                                            >
                                                <Typography
                                                    className={classes ? classes.navLinkTextDesktop : null}
                                                >
                                                    {link.text}
                                                </Typography>
                                            </NavLink>
                                    )
                                : null
                            }
                            <ToolbarAvatar
                                user={user}
                                login={() => this.props.googleSignIn()}
                                logout={() => this.props.signOut()}
                            />
                        </div>
                    </Toolbar>
                </AppBar>
                <MobileDrawer
                    isOpen={this.state.drawerOpen}
                    user={user}
                    closeDrawer={() => this.toggleDrawer(false)}
                    menuLinks={menuLinks}
                    logout={() => this.props.signOut() }
                    login={() => this.props.googleSignIn() }
                    selectedPath={this.props.location.pathname}
                />
            </div>
        );
    }
}

const mapStateToProps = state => ({
    user: state.auth.user
});

const mapDispatchToProps = dispatch => ({
    googleSignIn: () => dispatch(googleSignIn()),
    signOut: () => dispatch(signOut())
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(Header)));
