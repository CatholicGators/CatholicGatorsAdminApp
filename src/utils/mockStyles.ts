import { createMuiTheme } from '@material-ui/core'

export default (styles) => Object.keys(styles(createMuiTheme())).reduce((acc, curr) => ({...acc, [curr]: "test"}), {})