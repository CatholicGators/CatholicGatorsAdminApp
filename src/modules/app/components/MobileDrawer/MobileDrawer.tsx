import React from 'react';
import { NavLink } from "react-router-dom";

import { createStyles, withStyles, Theme } from '@material-ui/core';
import Divider from '@material-ui/core/Divider'
import Typography from '@material-ui/core/Typography';
import Avatar from '@material-ui/core/Avatar';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import PowerSettingsNew from '@material-ui/icons/PowerSettingsNew';
import ExitToApp from '@material-ui/icons/ExitToApp';
import CircularProgress from '@material-ui/core/CircularProgress';
import { MenuLink } from '../Header/Header';

export const styles = (theme: Theme) => createStyles({
    menuItems: {
        width: 250,
        marginBottom: 'auto'
    },
    header: {
        ...theme.mixins.toolbar,
        display: 'flex',
        alignItems: 'center',
        minHeight: '64px',
        padding: theme.spacing(1)
    },
    profileWrapper: {
        display: 'flex',
        alignItems: 'center',
    },
    avatar: {
        marginRight: theme.spacing(2)
    },
    headerName: {
        fontSize: '1rem',
        fontWeight: 'bold'
    },
    headerEmail: {
        fontSize: '0.75rem',
        color: '#727272'
    },
    navLink: {
        textDecoration: 'none',
        '&:visited': {
            color: 'inherit'
        }
    },
    progress: {
        color: theme.palette.secondary.main
    }
});

type Props = {
    classes: any,
    user: any,
    menuLinks: Array<MenuLink>,
    isOpen: boolean,
    selectedPath: string,
    closeDrawer: () => any,
    logout: () => any,
    login: () => any
}

export class MobileDrawer extends React.Component<Props> {
    render() {
        const { classes } = this.props
        return (
                <Drawer
                    open={this.props.isOpen}
                    onClose={() => this.props.closeDrawer()}
                >
                    <div className={classes ? classes.menuItems : null}>
                        <div className={classes ? classes.header : null}>
                            {this.getProfile()}
                        </div>
                        <Divider />
                        <List>
                            {this.getMenuLinks()}
                        </List>
                    </div>
                    {this.props.user ? 
                        <div>
                            <Divider />
                            <ListItem
                                id='logout-btn'
                                button
                                key='logout'
                                onClick={() => {
                                    this.props.closeDrawer()
                                    this.props.logout()
                                }}
                            >
                                <ListItemIcon>
                                    <ExitToApp/>
                                </ListItemIcon>
                                <ListItemText color='error' primary='Logout' />
                            </ListItem>
                        </div>
                    : null}
                </Drawer>
        );
    }

    getProfile() {
        const { classes, user } = this.props;

        switch(this.props.user) {
            case undefined:
                return <CircularProgress id='spinner' className={classes ? classes.progress : null} />
            case null:
                return null
            default:
                return (
                    <div className={classes ? classes.profileWrapper : null}>
                        <Avatar id='avatar' className={classes ? classes.avatar : null} src={user.photoURL} />
                        <div>
                            <Typography variant='h6' className={classes ? classes.headerName : null}>
                                {user.name}
                            </Typography>
                            <Typography variant='subtitle1' className={classes ? classes.headerEmail : null}>
                                {user.email}
                            </Typography>
                        </div>
                    </div>
                )
        }
    }

    getMenuLinks() {
        const { classes, user, menuLinks } = this.props;
        switch(user) {
            case undefined:
                return null;
            case null:
                return (
                    <ListItem
                        id='login-btn'
                        button
                        key='login'
                        onClick={() => {
                            this.props.closeDrawer()
                            this.props.login()
                        }}
                    >
                            <ListItemIcon>
                                <PowerSettingsNew/>
                            </ListItemIcon>
                            <ListItemText primary='Login' />
                    </ListItem>
                );
            default:
                return menuLinks
                            .filter(link => !link.needsAuthorization || user.isAdmin)
                            .map(link => 
                                <NavLink
                                    exact
                                    key={link.text}
                                    to={link.href}
                                    className={classes ? classes.navLink : null}
                                    onClick={() => this.props.closeDrawer()}
                                >
                                    <ListItem
                                        button
                                        key={link.text}
                                        selected={this.props.selectedPath == link.href}
                                    >
                                            <ListItemIcon>
                                                <link.icon/>
                                            </ListItemIcon>
                                            <ListItemText primary={link.text} />
                                    </ListItem>
                                </NavLink>
                            )
        }
    }
}

export default withStyles(styles)(MobileDrawer);
