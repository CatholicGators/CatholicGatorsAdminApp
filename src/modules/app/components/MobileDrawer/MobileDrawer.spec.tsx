import React from 'react'
import { shallow } from 'enzyme'
import { NavLink } from 'react-router-dom'
import toJson from 'enzyme-to-json'

import { menuLinks } from '../Header/Header'
import { MobileDrawer, styles } from './MobileDrawer'
import mockStyles from '../../../../utils/mockStyles'

describe('MobileDrawer', () => {
    let props, wrapper

    beforeEach(() => {
        props = {
            classes: mockStyles(styles),
            user: {
                isAdmin: true
            },
            menuLinks,
            isOpen: false,
            selectedPath: '/',
            closeDrawer: jest.fn(),
            logout: jest.fn(),
            login: jest.fn()
        }
        wrapper = shallow(<MobileDrawer {...props} />)
    })

    it('should match snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('closes the drawer when a NavLink is clicked on', () => {
        const listItem = wrapper.find(NavLink).first()

        listItem.simulate('click')

        expect(props.closeDrawer).toHaveBeenCalled()
    })

    describe('user is loading', () => {
        beforeEach(() => {
            wrapper.setProps({ user: undefined })
        })

        it('should only show a spinner', () => {
            expect(wrapper.exists('#spinner')).toBe(true)
            expect(wrapper.exists('#login-btn')).toBe(false)
            expect(wrapper.exists('#logout-btn')).toBe(false)
            expect(wrapper.exists('#avatar')).toBe(false)
        })

        it('renders no links', () => {
            menuLinks.forEach(link =>
                expect(wrapper.exists(`NavLink[to='${link.href}']`)).toBe(false)
            )
        })
    })

    describe('user is not logged in', () => {
        beforeEach(() => {
            wrapper.setProps({ user: null })
        })

        it('should only show the login btn', () => {
            expect(wrapper.exists('#spinner')).toBe(false)
            expect(wrapper.exists('#login-btn')).toBe(true)
            expect(wrapper.exists('#logout-btn')).toBe(false)
            expect(wrapper.exists('#avatar')).toBe(false)
        })

        it('renders no links', () => {
            menuLinks.forEach(link =>
                expect(wrapper.exists(`NavLink[to='${link.href}']`)).toBe(false)
            )
        })

        describe('login', () => {
            it('closes the drawer', () => {
                wrapper.find('#login-btn').simulate('click')

                expect(props.closeDrawer).toHaveBeenCalled()
            })

            it('calls googleSignIn()', () => {
                wrapper.find('#login-btn').simulate('click')

                expect(props.login).toHaveBeenCalled()
            })
        })
    })

    describe('user is logged in', () => {
        it('should show the logout btn and avatar', () => {
            expect(wrapper.exists('#spinner')).toBe(false)
            expect(wrapper.exists('#login-btn')).toBe(false)
            expect(wrapper.exists('#logout-btn')).toBe(true)
            expect(wrapper.exists('#avatar')).toBe(true)
        })

        it('renders all links when user is an admin', () => {
            menuLinks.forEach(link =>
                expect(wrapper.exists(`NavLink[to='${link.href}']`)).toBe(true)
            )
        })

        it('renders only unAuthorized links when user is not an admin', () => {
            wrapper.setProps({ user: { ...props.user, isAdmin: false } })

            menuLinks.forEach(link =>
                expect(wrapper.exists(`NavLink[to='${link.href}']`)).toBe(
                    !link.needsAuthorization
                )
            )
        })

        describe('logout', () => {
            it('closes the drawer', () => {
                wrapper.find('#logout-btn').simulate('click')

                expect(props.closeDrawer).toHaveBeenCalled()
            })

            it('calls logout()', () => {
                wrapper.find('#logout-btn').simulate('click')

                expect(props.logout).toHaveBeenCalled()
            })
        })
    })
})
