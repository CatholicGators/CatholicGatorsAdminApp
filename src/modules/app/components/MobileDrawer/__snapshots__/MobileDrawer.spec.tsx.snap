// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileDrawer should match snapshot 1`] = `
<WithStyles(ForwardRef(Drawer))
  onClose={[Function]}
  open={false}
>
  <div
    className="test"
  >
    <div
      className="test"
    >
      <div
        className="test"
      >
        <WithStyles(ForwardRef(Avatar))
          className="test"
          id="avatar"
        />
        <div>
          <WithStyles(ForwardRef(Typography))
            className="test"
            variant="h6"
          />
          <WithStyles(ForwardRef(Typography))
            className="test"
            variant="subtitle1"
          />
        </div>
      </div>
    </div>
    <WithStyles(ForwardRef(Divider)) />
    <WithStyles(ForwardRef(List))>
      <NavLink
        className="test"
        exact={true}
        key="Contact Form"
        onClick={[Function]}
        to="/"
      >
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="Contact Form"
          selected={true}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <FormatAlignLeftIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Contact Form"
          />
        </WithStyles(ForwardRef(ListItem))>
      </NavLink>
      <NavLink
        className="test"
        exact={true}
        key="My Contacts"
        onClick={[Function]}
        to="/my-contacts"
      >
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="My Contacts"
          selected={false}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <InboxIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="My Contacts"
          />
        </WithStyles(ForwardRef(ListItem))>
      </NavLink>
      <NavLink
        className="test"
        exact={true}
        key="Admin"
        onClick={[Function]}
        to="/admin"
      >
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="Admin"
          selected={false}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <VpnKeyIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Admin"
          />
        </WithStyles(ForwardRef(ListItem))>
      </NavLink>
    </WithStyles(ForwardRef(List))>
  </div>
  <div>
    <WithStyles(ForwardRef(Divider)) />
    <WithStyles(ForwardRef(ListItem))
      button={true}
      id="logout-btn"
      key="logout"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <ExitToAppIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        color="error"
        primary="Logout"
      />
    </WithStyles(ForwardRef(ListItem))>
  </div>
</WithStyles(ForwardRef(Drawer))>
`;
